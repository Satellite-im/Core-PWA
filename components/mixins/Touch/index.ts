export const Touch = {
  created: () => {
    // can add functions here on start-up if needed
  },
  methods: {
    /**
     * @method sidebarSwipeHandler DocsTODO
     * @description
     * @param currThis
     * @returns
     * @example
     */
    sidebarSwipeHandler(currThis: any) {
      return function (direction: string) {
        const showSettings = currThis.$store.state.ui.showSettings
        if (direction === 'left' && currThis.$device.isMobile && !showSettings) {
          currThis.sidebar = false
        }
        if (direction === 'right' && currThis.$device.isMobile && !showSettings) {
          currThis.sidebar = true
        }
      }
    },
  },
}

export default Touch