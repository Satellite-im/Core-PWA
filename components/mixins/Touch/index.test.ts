import * as index from '~/components/mixins/Touch/index'

describe('index.Touch.created', () => {
  test('0', () => {
    const localSpy = jest.spyOn(index.Touch, 'created')
    const result: any = index.Touch.created()

    expect(localSpy).toHaveBeenCalled()
    expect(result).toBe(undefined)
  })
})

describe('index.Touch.methods.sidebarSwipeHandler human created', () => {
  test('showSettings false and left direction', () => {
    const local = {
      $store: {
        state: {
          ui: {
            showSettings: false,
          },
        },
      },
      $device: {
        isMobile: true,
      },
      sidebar: false,
    }
    const givenFunction: any = index.Touch.methods.sidebarSwipeHandler(local)
    givenFunction('left')
    expect(local.sidebar).toBeFalsy()
  })

  test('showSettings false and right direction', () => {
    const local = {
      $store: {
        state: {
          ui: {
            showSettings: false,
          },
        },
      },
      $device: {
        isMobile: true,
      },
      sidebar: false,
    }
    const givenFunction: any = index.Touch.methods.sidebarSwipeHandler(local)
    givenFunction('right')
    expect(local.sidebar).toBeTruthy()
  })

  test('showSettings true and left direction', () => {
    const local = {
      $store: {
        state: {
          ui: {
            showSettings: true,
          },
        },
      },
      $device: {
        isMobile: true,
      },
      sidebar: false,
    }
    const givenFunction: any = index.Touch.methods.sidebarSwipeHandler(local)
    givenFunction('left')
    expect(local.sidebar).toBeFalsy()
  })

  test('showSettings true and right direction', () => {
    const local = {
      $store: {
        state: {
          ui: {
            showSettings: true,
          },
        },
      },
      $device: {
        isMobile: true,
      },
      sidebar: false,
    }
    const givenFunction: any = index.Touch.methods.sidebarSwipeHandler(local)
    givenFunction('right')
    expect(local.sidebar).toBeFalsy() // Will be false because no mutation is done to the local, because showSettings is not false.
  })
})
