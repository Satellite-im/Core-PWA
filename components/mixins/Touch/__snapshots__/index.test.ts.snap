// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.Touch.created 0 1`] = `undefined`;
